### PR TITLE
Rotation align (manual) operator fixes

### PR DESCRIPTION
**Pull request by mathturtle**
This branch makes several improvements to the rotation align operator.  

* merges the operators that were generated (a shift and rotate) into one rotation align operator (a new operator).

* adds the ability for Python operators to use custom UIs for their edit dialogs, by specifying the key "widget": "abcd" in the operator's JSON file.

* the rotation align operator is given one of these custom widgets (the existing RotationAlignWidget is modified to fit the new API and registered as a custom widget for JSON based python operators).

* the dialog that allows editing of already created python operators is given tabs, one for the existing text area to edit the python script and one to show the editor widget for the operator (the custom widget if applicable, or the widget we show in the properties panel if not).

* when creating python operators that have JSON-based UIs, the EditOperatorDialog is used similar to many of the non-python operators.  This gives a consistent UI for creating and editing these operators which I think is an improvement over the 'you can only edit arguments when creating and only edit the script later' dialogs that we had before.

### CODE REVIEW

**Comment by mathturtle** [comment]
The new create/edit operator dialog: you can edit the script and/or the arguments here.  It defaults to the tab for the arguments.

new_create_operator

**Comment by mathturtle** [comment]
An example of the custom widget for the rotate align operator in this dialog.  It should work the same way it did (and does in my testing...), except you can get it back by double clicking the operator now.

rotatealigndialog

**Comment by ercius** [comment]
Looks great!

One thing missing from this UI is the ability to change the
brightness/contrast of the reconstructed slices. I think one setting for
all three images is sufficient. That could also include the color map (one
setting for all three).

**Comment by cquammen** [comment]
I think this looks good, and adds a crucial capability to the JSON parameter description format (specifying custom widgets). Implementation looks solid, and the few operators I have tried appear to operate as expected.

I think @ercius 's request for the brightness/contrast could be addressed in a separate merge request.

**Comment by cryos** [comment]
Apparently the GitHub web editor failed, @mathturtle I suggest deleting the useless commit I added with the web editor, resolving the conflict, and then we can get this branch merged.

File: tomviz/EditPythonOperatorWidget.ui [hunk 1]
```diff
@@ -105,7 +97,21 @@
...
     utils.set_array(dataset, result)</string>
-     </property>
+         </property>
+        </widget>
+       </item>
+      </layout>
+     </widget>
+     <widget class="QWidget" name="tab_2">
+      <attribute name="title">
+       <string>Set Argument Values</string>
```
[the remark this answers is not in the document]
Comment (mathturtle) [reply tomviz/EditPythonOperatorWidget.ui]: I updated this with my rebase.

**Comment by mathturtle** [comment]
Rebased to fix conflicts (fortunately just includes in `Behaviors.cxx`, nothing complicated).

**Pull request merged**